fix(core): unify account linking trust checks across SSO and core OAuth (#6509)
better-auth/better-auth: docs/content/docs/concepts/users-accounts.mdx

});
```

#### Existing User Auto-Linking Policy

The `existingUserMode` option controls when OAuth/SSO providers can auto-link to an existing user with the same email address. This provides fine-grained control over account linking security.

```ts title="auth.ts"
export const auth = betterAuth({
    account: {
        accountLinking: {
            enabled: true,

>             existingUserMode: "trusted_providers_only", // recommended

Reviewer: This really sounds like a policy to me. What do you think of renaming this option to `existingUserPolicy`?

        }
    },
});
```

Available modes:

- **`"never"`**: Never auto-link. Users must explicitly link accounts (e.g., via a signed-in "link account" flow).

> - **`"trusted_providers_only"`** (recommended): Only auto-link if the provider passes a trust check. This means the provider must be listed in `trustedProviders`. For SSO providers, domain verification also counts as a trust signal.

Reviewer: I wonder if we could rephrase this in a way that is not specific to SSO and then we can further elaborate on the SSO plugin docs (which I think we already do).

```suggestion
- **`"trusted_providers_only"`** (recommended): Only auto-link if the provider passes a trust check. This means the provider must be listed in `trustedProviders` or the associated domain has been verified.
```

> - **`"email_match_any"`** (default): Legacy behavior for backward compatibility. Auto-link when either the provider is trusted OR the provider returns `email_verified: true`.

Author: The policy does two things:
Links if provider is in trustedProviders (same as trusted_providers_only)
Also links if provider returns email_verified: true

So the name should convey "we also trust providers that verify email".

Maybe:
verified_email_trusted - "a verified email is treated as trusted"
implicit_email_trust - "implicit trust via email"

Leaning to verified_email_trusted

> - **`"email_match_any"`** (default): Legacy behavior for backward compatibility. Auto-link when either the provider is trusted OR the provider returns `email_verified: true`.

Reviewer: `verified_email_trusted ` is a great name!
